.github/workflows/main.yml

```diff
@@ -0,0 +1,74 @@
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build-toolchain:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        # we don't need submodules for building the toolchain
```

Comment:
Sure, the reason I did it this way was to shave off a little time, because even if there's a cache hit in the end, the checkout always runs because `build-toolchain.sh` becomes part of the cache key. I can match the checkout used for `build-sdk` though.